src/base/coordinate-grid-mixin.js

```diff
@@ -429,6 +433,36 @@ export class CoordinateGridMixin extends ColorMixin(MarginMixin) {
         return this;
     }
 
+    /**
+     * Gets or sets whether the chart should be drawn with a top axis instead of a bottom axis. When
+     * used with a chart in a composite chart, allows both top and bottom X axes to be shown on a
+     * chart.
+     * @param {Boolean} [useTopXAxis=false]
+     * @returns {Boolean|CoordinateGridMixin}
+     */
+    useTopXAxis (useTopXAxis) {
+        if (!arguments.length) {
+            return this._useTopXAxis;
+        }
+
+        // We need to warn if value is changing after self._yAxis was created
+        if (this._useTopXAxis !== useTopXAxis && this._xAxis) {
+            logger.warn('Value of useTopXAxis has been altered, after xAxis was created. ' +
+                'You might get unexpected yAxis behavior. ' +
+                'Make calls to useTopXAxis sooner in your chart creation process.');
+        }
+
+        this._useTopXAxis = useTopXAxis;
+
+        // adjust default top margin to make room for the labels
+        if (this._useTopXAxis && this.margins().top < 30)
+        {
+            this.margins().top = 30;
+        }
```

Comment:
We don't generally adjust the margins automatically, because it gets very complicated very quickly with all the interacting features.

I think it's better to change the example to set the margins. 

Right now the example has a big margin at the bottom which it is not using.

<img width="412" alt="image" src="https://user-images.githubusercontent.com/1366709/109509138-946b8e00-7a6e-11eb-85aa-330e3d37283e.png">

Reply:
Makes perfect sense!